mozdownload/scraper.py

```diff
@@ -892,7 +891,13 @@ def get_build_info_for_index(self, build_index=None):
 
         # If no index has been given, set it to the last build of the day.
         if build_index is None:
-            build_index = len(parser.entries) - 1
+            # Find the most recent non-empty entry.
+            build_index = len(parser.entries)
+            for build in reversed(parser.entries):
+                build_index -= 1
+                if not build_index or self.is_build_dir(build):
+                    break
+
```

Comment:
> But 0 is a valid index anyway. You don't want to assume that the directory at index 0 is OK. If the directory at index 0 is not a good build directory, this will return it anyway.

I just checked this myself given that I haven't gotten a response from @EricRahm yet on this PR or issue. So we indeed return the oldest build folder for tinderbox builds but a download will fail with a NotFoundError anyway. It's only a matter of the message we print. That's all. We could improve that later by printing the platform build folder instead of the first build folder under it.